controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -152,30 +158,40 @@ func NewClient(ctx context.Context, config ClientConfiguration) (*Client, error)
 		return nil, errors.Wrap(err, "unable to create etcd client")
 	}
 
-	client, err := newEtcdClient(ctx, etcdClient)
+	callTimeout := config.CallTimeout
+	if callTimeout == 0 {
+		callTimeout = DefaultCallTimeout
+	}
+
+	client, err := newEtcdClient(ctx, etcdClient, callTimeout)
 	if err != nil {
 		closeErr := etcdClient.Close()
 		return nil, errors.Wrap(kerrors.NewAggregate([]error{err, closeErr}), "unable to create etcd client")
 	}
 	return client, nil
 }
 
-func newEtcdClient(ctx context.Context, etcdClient etcd) (*Client, error) {
+func newEtcdClient(ctx context.Context, etcdClient etcd, callTimeout time.Duration) (*Client, error) {
+
```

Comment:
```suggestion
```
(linter finding)

Reply:
🙈 Thanks! Pushed up the fix and re-ran the linter this time, should be all good!